docs/development/high-availability.md

```diff
@@ -47,22 +47,23 @@ Concretely, all seed system components should respect the following conventions:
     ```yaml
     spec:
       topologySpreadConstraints:
-      - maxSkew: 1
-        topologyKey: kubernetes.io/hostname
+      - topologyKey: kubernetes.io/hostname
+        minDomains: 3 # lower value of max replicas or 3
+        maxSkew: 1
         whenUnsatisfiable: ScheduleAnyway
         matchLabels: ...
     ```
 
-    Hence, the node spread is done on best-effort basis only.
+    `minDomains` is set when failure tolerance is `node` or annotation `high-availability-config.resources.gardener.cloud/host-spread="true"` is given. Otherwise, e.g. for shoots with failure tolerance type `zone`, the node spread is done on best-effort basis only.
```

Comment:
What is the reasoning to not also do it for failure tolerance type `zone`?

Reply:
I think there is simply no benefit since we already spread across zones, and thus the desired amount of nodes (zone shoots are only scheduled on seeds >= 3 zones). Still, I changed it for the sake of code and doc simplicity.